nl_server/tests/lib/nl_embeddings_test.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Embeddings (in nl_embeddings.py)."""
+
+from diskcache import Cache
+from lib.nl_embeddings import Embeddings
+from nl_server import nl_cache_path, nl_embeddings_cache_key
+from parameterized import parameterized
+
+import unittest
+
+
+class TestEmbeddings(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(cls) -> None:
+
+    # Look for the Embeddings model in the cache if it exists.
+    cache = Cache(nl_cache_path)
+    cache.expire()
+    cls.nl_embeddings = cache.get(nl_embeddings_cache_key)
+    if not cls.nl_embeddings:
+      print(
+          "Could not load the embeddings from the cache for these tests. Loading a new embeddings object."
+      )
+      # Using the default NER model.
+      cls.nl_embeddings = Embeddings()
```

Comment:
I wonder shall we always load embeddings instead of reading from Cache. The cache is mostly for faster reload, use it for testing may introduce discrepency.

Reply:
Well, if we don't do this then the local testing situation becomes pretty painful as well (it will download a lot of stuff and do it every time we want to test locally). If there ends up being a discrepancy between the local and remote tests, the fix is easy: empty local cache and retry on local. So I think this is Ok for now?

Reply:
that sg!